crates/rolldown/src/dev/bundle_coordinator.rs

```diff
@@ -350,20 +356,29 @@ impl BundleCoordinator {
           }
         } else {
           let schedule_result = self.schedule_build_if_stale().await?;
-          Ok(schedule_result.map(|ret| ret.future))
+          Ok(schedule_result.map(|ret| EnsureLatestBundleOutputReturn {
+            future: ret.future,
+            is_ensure_latest_bundle_output_future: false,
+          }))
         }
       }
       CoordinatorState::FullBuildInProgress | CoordinatorState::InProgress => {
         tracing::trace!("[BundleCoordinator] found running build and end ensuring");
         // If there's a build running, return its future
-        Ok(Some(self.current_bundling_future.clone().unwrap()))
+        Ok(Some(EnsureLatestBundleOutputReturn {
+          future: self.current_bundling_future.clone().unwrap(),
+          is_ensure_latest_bundle_output_future: false,
+        }))
       }
       CoordinatorState::FullBuildFailed => {
         // Clear all queued tasks and schedule a new full build
         self.queued_tasks.clear();
         self.queued_tasks.push_back(TaskInput::FullBuild);
         let schedule_result = self.schedule_build_if_stale().await?;
-        Ok(schedule_result.map(|ret| ret.future))
+        Ok(schedule_result.map(|ret| EnsureLatestBundleOutputReturn {
+          future: ret.future,
+          is_ensure_latest_bundle_output_future: true,
+        }))
       }
       CoordinatorState::Failed => {
         todo!("how we're gonna handle this?")
```

Comment:
I think we can treat this the same as `CoordinatorState::Idle`.
I also think `CoordinatorState::InProgress` should be treated the same as `CoordinatorState::Idle`. HMR rebuild does not update the bundle output, so we need to schedule a new full bundle task.